## FILM DATA ANALYSIS FOR SUCCESSFUL LAUNCH INTO THE INDUSTRY

## INTRODUCTION 
Major corporations are increasingly investing in original content creation, our company is preparing to launch a movie studio to capitalize on the growing entertainment market. With limited prior experience in film production, this project is designed to deliver data-driven insights that will guide our entry into the industry.

We will be analyzing comprehensive datasets from across the film ecosystem with the aim to uncover patterns of success. We will site genres,creators and markets that demonstrate high potential for prophitability and critical acclaim.

## BUSINESS UNDERSTANDING
### Stakeholders
1. *Chief operating officer* - He will be providing the vision and oversee the project
2. *Creative and productions team department head* - Responsible for bringing the content to life and responsible for making decisions that impact quality
3. *Finace team* - Responsible for funding the projects and manage the income from the same
4. *Marketing team* - Responsible for marketing the new business
5. *Data and strategy team* - They guide on risk mitigation

### Business value
This project projects key benefits to our organization. Some of the key benefits include;
1. *Data pivoted decision making* - Reduced risk and maximizing on ROI by backing projects with data
2. *Operation Efficiency* - Allows producers to focus on high potential projects and shprten decision making cycles
3. *Market differentiation* - Using data driven decision sets the organization apart from the crowded industry.This give the organization a competitive edge
4. *Stakeholder confidence* - The project will build confidence with investors and partners , hence attracting more investors

### Project goal
This project aims to answer questions below;
1.  Which film genres receive the best critical reception?
2.	What genres offer the highest return on investment (ROI)?
3.	Which languages and production markets are most commonly associated with popular, high-grossing films?
4.  Are there directors whose films consistently perform well internationally?

## Data Sources
To generate reliable insights, we are going to integrate data from multiple authoratative sources. They include:

*Box Office Mojo* - Worldwide gross earnings and box office breakdowns

*Rotten Tomatoes* - Critic scores, audience ratings, review sentiment 

*IMDb* - Cross-referenced metadata: genres, actors, directors

*TheMovieDB (TMDB*)- Metadata: genres, languages, production countries, popularity

*The Numbers*- Financial data: budgets, revenues, profit estimates

Together, these datasets offer a comprehensive view of the global film landscape.

## Conclusion
In film industry ,creative success must align with finacial perfomance. Our entry to the industry therefore requires strategic clarity.

This project positions data as a foundational tool in our studio’s decision-making. By analyzing trends in genre performance, critical reception, financial returns, and global market behavior, we aim to uncover actionable insights that will guide which projects to greenlight. The following data analysis serves as the backbone of this approach—transforming industry complexity into clear, evidence-based recommendations for sustainable growth in entertainment production.

## Data Exploration

#### *Importing the Data*


It's essential that we conduct a comprehensive examination of all our datasets and database tables. The data sources we are working with include:

- IMDB

- TheMovieDB

- Rotten Tomatoes Movies

- Rotten Tomatoes Critic Reviews

- Box Office Mojo

- The Numbers

- im.db (SQLite database)

Data exploration helps us understand the structure, quality, and patterns within the data. By examining things like missing values, duplicates, and relationships between variables, we can detect potential issues early. This allows us to plan effective cleaning and transformation steps, laying a solid foundation for accurate and meaningful analysis. 

Below are the steps followed:


In [141]:
# Import necessary libraries
import pandas as pd
import numpy as np
import sqlite3
import matplotlib.pyplot as plt
import seaborn as sns

In [142]:
# Establish a connection to the database
conn = sqlite3.connect('Data/im.db')

# Check available tables (optional, good for understanding structure)
pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table';", conn)

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [143]:
# Close connection
conn.close()

In [144]:
bom_df = pd.read_csv('Data/bom.movie_gross.csv.gz')
bom_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [145]:
rt_rev_df = pd.read_csv('Data\\rotten_tomatoes_critic_reviews.csv')
rt_rev_df.head()

rotten_tomatoes_link      critic_name  top_critic           publisher_name  \
0            m/0814255  Andrew L. Urban       False           Urban Cinefile   
1            m/0814255    Louise Keller       False           Urban Cinefile   
2            m/0814255              NaN       False      FILMINK (Australia)   
3            m/0814255     Ben McEachen       False  Sunday Mail (Australia)   
4            m/0814255      Ethan Alter        True       Hollywood Reporter   

  review_type review_score review_date  \
0       Fresh          NaN  2010-02-06   
1       Fresh          NaN  2010-02-06   
2       Fresh          NaN  2010-02-09   
3       Fresh        3.5/5  2010-02-09   
4      Rotten          NaN  2010-02-10   

                                      review_content  
0  A fantasy adventure that fuses Greek mythology...  
1  Uma Thurman as Medusa, the gorgon with a coiff...  
2  With a top-notch cast and dazzling special eff...  
3  Whether audiences will get behind The Lightnin...  
4  What's really lacking in The Lightning Thief i...

In [146]:
rt_df = pd.read_csv('Data\\rotten_tomatoes_movies.csv')

rt_df.head()

rotten_tomatoes_link  \
0                              m/0814255   
1                              m/0878835   
2                                   m/10   
3                 m/1000013-12_angry_men   
4  m/1000079-20000_leagues_under_the_sea   

                                         movie_title  \
0  Percy Jackson & the Olympians: The Lightning T...   
1                                        Please Give   
2                                                 10   
3                    12 Angry Men (Twelve Angry Men)   
4                       20,000 Leagues Under The Sea   

                                          movie_info  \
0  Always trouble-prone, the life of teenager Per...   
1  Kate (Catherine Keener) and her husband Alex (...   
2  A successful, middle-aged Hollywood songwriter...   
3  Following the closing arguments in a murder tr...   
4  In 1866, Professor Pierre M. Aronnax (Paul Luk...   

                                   critics_consensus content_rating  \
0  Though it may seem like just another Harry Pot...             PG   
1  Nicole Holofcener's newest might seem slight i...              R   
2  Blake Edwards' bawdy comedy may not score a pe...              R   
3  Sidney Lumet's feature debut is a superbly wri...             NR   
4  One of Disney's finest live-action adventures,...              G   

                                              genres          directors  \
0  Action & Adventure, Comedy, Drama, Science Fic...     Chris Columbus   
1                                             Comedy  Nicole Holofcener   
2                                    Comedy, Romance      Blake Edwards   
3                                    Classics, Drama       Sidney Lumet   
4           Action & Adventure, Drama, Kids & Family  Richard Fleischer   

                                      authors  \
0  Craig Titley, Chris Columbus, Rick Riordan   
1                           Nicole Holofcener   
2                               Blake Edwards   
3                               Reginald Rose   
4                                 Earl Felton   

                                              actors original_release_date  \
0  Logan Lerman, Brandon T. Jackson, Alexandra Da...            2010-02-12   
1  Catherine Keener, Amanda Peet, Oliver Platt, R...            2010-04-30   
2  Dudley Moore, Bo Derek, Julie Andrews, Robert ...            1979-10-05   
3  Martin Balsam, John Fiedler, Lee J. Cobb, E.G....            1957-04-13   
4  James Mason, Kirk Douglas, Paul Lukas, Peter L...            1954-01-01   

   ...      production_company  tomatometer_status tomatometer_rating  \
0  ...        20th Century Fox              Rotten               49.0   
1  ...  Sony Pictures Classics     Certified-Fresh               87.0   
2  ...             Waner Bros.               Fresh               67.0   
3  ...    Criterion Collection     Certified-Fresh              100.0   
4  ...                  Disney               Fresh               89.0   

  tomatometer_count  audience_status  audience_rating audience_count  \
0             149.0          Spilled             53.0       254421.0   
1             142.0          Upright             64.0        11574.0   
2              24.0          Spilled             53.0        14684.0   
3              54.0          Upright             97.0       105386.0   
4              27.0          Upright             74.0        68918.0   

   tomatometer_top_critics_count  tomatometer_fresh_critics_count  \
0                             43                               73   
1                             44                              123   
2                              2                               16   
3                              6                               54   
4                              5                               24   

   tomatometer_rotten_critics_count  
0                                76  
1                                19  
2                                 8  


In [147]:
tmdb_df = pd.read_csv('Data/tmdb.movies.csv.gz')
tmdb_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [148]:
tn_df = pd.read_csv('Data/tn.movie_budgets.csv.gz')
tn_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

Data Cleaning and Preparation
Before performing any meaningful analysis, it's crucial to thoroughly prepare and clean the data. Although our initial data exploration verified that the datasets are of high quality, they originate from diverse sources. As part of the preparation process, we need to filter key columns, standardize field names, address duplicates, and merge datasets when required.

Preparing Rotten Tomatoes Ratings and Genres Data
To analyze how movie genres and ratings relate to potential success, we will process and refine the Rotten Tomatoes Movies dataset. This dataset contains comprehensive metadata from the Rotten Tomatoes website, covering a diverse collection of films. The information it provides will be crucial for addressing our business questions about audience preferences and genre-based trends

Ratings and Genres Data Extraction
For our analysis of movie success and audience perception, we concentrate on four critical columns from the dataset:

movie_title: The official title of the film as recorded on Rotten Tomatoes.

genres: A comma-separated string categorizing each movie into one or more genres (e.g., Drama, Comedy, Action).

audience_rating: The average audience score, typically scaled from 0 to 100.

tomatometer_rating: The percentage of favorable professional critic reviews.

During data preparation, special attention is required for movies classified under multiple genres to ensure proper parsing and accurate analysis.

In [149]:
# Extracting the relevant columns
df_rt = rt_df[["movie_title","genres","audience_rating","tomatometer_rating"]].copy()

# Convert the comma-separated string into a list
df_rt['genres'] = df_rt['genres'].str.split(', ')

# Exploding the genres
rt_movies_exploded = df_rt.explode('genres').reset_index(drop=True)
rt_movies_exploded.head()

movie_title  \
0  Percy Jackson & the Olympians: The Lightning T...   
1  Percy Jackson & the Olympians: The Lightning T...   
2  Percy Jackson & the Olympians: The Lightning T...   
3  Percy Jackson & the Olympians: The Lightning T...   
4                                        Please Give   

                      genres  audience_rating  tomatometer_rating  
0         Action & Adventure             53.0                49.0  
1                     Comedy             53.0                49.0  
2                      Drama             53.0                49.0  
3  Science Fiction & Fantasy             53.0                49.0  
4                     Comedy             64.0                87.0

We perform an operation to split the genres column, creating separate rows for each genre entry. This process converts the one-to-many relationships between movies and their genres into simpler one-to-one relationships, facilitating genre-specific analysis. Following this transformation, we reset the DataFrame index to maintain proper data organization and ensure clean dataset structure.

In [150]:
# Comparing the number of rows for main df and the exploded df 
print(f"main: {rt_df.shape[0]} rows")
print(f"exploded: {rt_movies_exploded.shape[0]} rows")

main: 17712 rows
exploded: 39388 rows


By comparing row counts between the original and transformed DataFrames, we can validate the successful separation of multi-genre entries. The expected increase in row count confirms that each genre has been properly isolated into distinct records

In [151]:
# Seeing if the number of unique movie_title values matches with the main
print(f"exploded: {len(rt_movies_exploded['movie_title'].unique())} unique values")
print(f"main df: {len(rt_df['movie_title'].unique())} unique values")

exploded: 17106 unique values
main df: 17106 unique values


We confirm data integrity by checking that unique movie title counts match between datasets before and after transformation, ensuring no films were duplicated or lost during genre separation

Data Cleaning
Our preliminary analysis verified that the key columns of interest; genres, audience_rating, and tomatometer rating to maintain complete data integrity with no duplicate entries. Each field contains distinct measurements, confirming proper data structure.
We did identify a negligible proportion of missing values, representing only 0-1% of records across these columns. Given this minimal occurrence, we can confidently address these gaps through removing rows without compromising the dataset's statistical reliability or analytical value.


In [152]:
# Checking if missing values align with data exploration findings
(rt_movies_exploded.isna().sum()/len(rt_movies_exploded)*100)

movie_title           0.000000
genres                0.048238
audience_rating       1.241495
tomatometer_rating    0.233574
dtype: float64

We confirm that the missing values align with our previous findings. We proceed with dropping the rows containing missing values.

In [153]:
# Dropping rows with missing values under 'genres', 'audience_rating', and 'tomatometer_rating'
rt_movies_exploded_cleaned = (
    rt_movies_exploded
    .dropna(subset=["genres", "audience_rating", "tomatometer_rating"])
    .reset_index(drop=True)
)

# Checking if there are any missing values left
rt_movies_exploded_cleaned.isna().sum()

movie_title           0
genres                0
audience_rating       0
tomatometer_rating    0
dtype: int64

Checking for Outliers
We will use the Interquartile Range (IQR) method to check for outliers:

IQR = Q3 - Q1
Any data point below Q1 - 1.5 * IQR or above Q3 + 1.5 * IQR is considered an outlier.

We will apply this logic to audience_rating and tomatometer_rating

In [154]:
# Checking for outliers
def check_outliers(q1,q3,min,max):
    iqr=q3-q1
    upper_bound=q3+(1.5*iqr)
    lower_bound=q1-(1.5*iqr)
    if (min>lower_bound) & (max<upper_bound):
     return("There are no outliers")
    else:
     return("There are outliers")
 
# Checking in the audience_rating column
print(f"{check_outliers(q1=46,q3=78,min=0,max=100)} in the audience rating column")

# Checking in the tomatometer_rating column
print(f"{check_outliers(q1=40,q3=86,min=0,max=100)} in the tomatometer rating column")

There are no outliers in the audience rating column
There are no outliers in the tomatometer rating column


Standardization
This helps to avoid inconsistencies during grouping, searching, or filtering. (e.g., "Drama" and "drama" would otherwise be treated differently). We standardize text data by converting movie_title and genres columns to lowercase.

In [155]:
# Making sure to work on a copy to avoid the SettingWithCopyWarning
rt_movies_exploded_cleaned = rt_movies_exploded_cleaned.copy()

# Standardizing columns to lowercase
rt_movies_exploded_cleaned["genres"] = rt_movies_exploded_cleaned["genres"].str.lower()
rt_movies_exploded_cleaned["movie_title"] = rt_movies_exploded_cleaned["movie_title"].str.lower()

rt_movies_exploded_cleaned.head()

movie_title  \
0  percy jackson & the olympians: the lightning t...   
1  percy jackson & the olympians: the lightning t...   
2  percy jackson & the olympians: the lightning t...   
3  percy jackson & the olympians: the lightning t...   
4                                        please give   

                      genres  audience_rating  tomatometer_rating  
0         action & adventure             53.0                49.0  
1                     comedy             53.0                49.0  
2                      drama             53.0                49.0  
3  science fiction & fantasy             53.0                49.0  
4                     comedy             64.0                87.0

Grouping Ratings by Genre
To identify trends in how different genres perform, we grouped the movies by category and calculated average ratings from both audiences and critics. This gives us the big picture on which types of films viewers and reviewers prefer, rather than getting stuck on individual movie scores.

In [156]:
# Grouping by genres and aggregating the mean of audience_rating and tomatometer_rating
genre_rating_df = rt_movies_exploded_cleaned.groupby('genres').agg({
    'audience_rating': 'mean',
    'tomatometer_rating': 'mean'
}).reset_index()

# Formatting the ratings to show only one decimal place
genre_rating_df['audience_rating'] = genre_rating_df['audience_rating'].round(1)
genre_rating_df['tomatometer_rating'] = genre_rating_df['tomatometer_rating'].round(1)

genre_rating_df

genres  audience_rating  tomatometer_rating
0          action & adventure             57.9                54.2
1                   animation             65.7                64.1
2               anime & manga             75.0                75.4
3   art house & international             66.5                71.0
4                    classics             71.0                78.4
5                      comedy             58.4                55.1
6                 cult movies             55.8                60.5
7                 documentary             73.3                80.9
8                       drama             63.1                63.2
9        faith & spirituality             70.3                63.9
10              gay & lesbian             61.5                57.6
11                     horror             46.4                51.6
12              kids & family             60.3                55.7
13  musical & performing arts             70.9                71.0
14         mystery & suspense             54.0                55.3
15                    romance             63.4                60.5
16  science fiction & fantasy             56.7                55.9
17           special interest             72.0                77.7
18           sports & fitness             73.4                71.0
19                 television             66.8                72.4
20                    western             64.6                68.5

In [157]:
# Confirming no genres were lost after aggregation
len(rt_movies_exploded_cleaned['genres'].value_counts()) == len(genre_rating_df)

True

Preparing Genres and ROI Data
In this section, we will focus on preparing two key components: genres and Return on Investment (ROI). Both are essential for understanding the financial success and audience preferences of movies
We'll be working with two key datasets:

Movie budgets (tn.movie_budgets): Shows what each film cost to make (Production Budget), how much it earned globally (Worldwide Gross), and its Return on Investment (ROI).

Movie genres (rt_movies): Tells us what categories each film falls into, like Action, Comedy, or Drama.

Why does this matter? It help us understand what kinds of stories audiences love most. ROI shows us which films gave studios value for money. By connecting these dots, we can figure out which types of movies are both popular AND profitable.

Merging Appropriate Tables
We will merge the tn.movie_budgets dataset (which contains production budgets and worldwide gross revenue) with the rt_movies dataset (which includes genre information) to create a consolidated table that includes both genres and ROI for each movie.

In [158]:
# Prepare movie titles for better matching (remove special characters, lower case)
tn_df['movie'] = tn_df['movie'].str.strip().str.lower()
rt_df['movie_title'] = rt_df['movie_title'].str.strip().str.lower()

# Merge the datasets on the appropriate columns
merged_df = tn_df.merge(rt_df[['movie_title', 'genres']], left_on='movie', right_on='movie_title', how='left')

# Drop duplicate movie_title column if you want
merged_df = merged_df.drop(columns=['movie_title'])

merged_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       avatar   
1   2  May 20, 2011  pirates of the caribbean: on stranger tides   
2   3   Jun 7, 2019                                 dark phoenix   
3   4   May 1, 2015                      avengers: age of ultron   
4   5  Dec 15, 2017            star wars ep. viii: the last jedi   

  production_budget domestic_gross worldwide_gross  \
0      $425,000,000   $760,507,625  $2,776,345,279   
1      $410,600,000   $241,063,875  $1,045,663,875   
2      $350,000,000    $42,762,350    $149,762,350   
3      $330,600,000   $459,005,868  $1,403,013,963   
4      $317,000,000   $620,181,382  $1,316,721,747   

                                              genres  
0  Action & Adventure, Comedy, Mystery & Suspense...  
1  Action & Adventure, Comedy, Science Fiction & ...  
2  Action & Adventure, Drama, Science Fiction & F...  
3      Action & Adventure, Science Fiction & Fantasy  
4                                                NaN

ROI Computation
ROI is a key financial metric that helps us assess the profitability of a film relative to its production cost. By calculating the ROI for each movie, we better understand which films have been the most financially successful and use that insight to inform future production decisions.

The calculation of ROI will be based on the formula:

ROI = (Worldwide Gross - Production Budget) ÷ Production Budget × 100

In [159]:
# Clean up the dollar signs and commas and convert columns to numeric
merged_df['production_budget'] = merged_df['production_budget'].replace([r'[\$,]'], '', regex=True).astype(float)
merged_df['worldwide_gross'] = merged_df['worldwide_gross'].replace([r'[\$,]'], '', regex=True).astype(float)

# Calculate ROI
merged_df['ROI (%)'] = ((merged_df['worldwide_gross'] - merged_df['production_budget']) / merged_df['production_budget']) * 100

# Round ROI to nearest whole number
merged_df['ROI (%)'] = merged_df['ROI (%)'].round()

# Group by 'movie' and calculate the mean ROI (though each movie is unique here)
roi_grouped = merged_df.groupby('movie')[['ROI (%)']].mean().reset_index()

# Final table: Comparison between genres and the grouped movie ROI, while keeping the production_budget and worldwide_gross columns
final_table = merged_df[['genres', 'movie', 'production_budget', 'worldwide_gross']].merge(roi_grouped, on='movie')

final_table.head()

genres  \
0  Action & Adventure, Comedy, Mystery & Suspense...   
1  Action & Adventure, Comedy, Science Fiction & ...   
2  Action & Adventure, Drama, Science Fiction & F...   
3      Action & Adventure, Science Fiction & Fantasy   
4                                                NaN   

                                         movie  production_budget  \
0                                       avatar        425000000.0   
1  pirates of the caribbean: on stranger tides        410600000.0   
2                                 dark phoenix        350000000.0   
3                      avengers: age of ultron        330600000.0   
4            star wars ep. viii: the last jedi        317000000.0   

   worldwide_gross  ROI (%)  
0     2.776345e+09    553.0  
1     1.045664e+09    155.0  
2     1.497624e+08    -57.0  
3     1.403014e+09    324.0  
4     1.316722e+09    315.0

Handling Missing Values

In [160]:
final_table.isna().sum()

genres               1264
movie                   0
production_budget       0
worldwide_gross         0
ROI (%)                 0
dtype: int64

Since the genres column has 1264 missing values and is crucial for our analysis, we will drop these rows as they do not provide useful information for our objectives.

In [161]:
final_table.dropna(subset=['genres'], inplace=True)

Expanding Movies into Individual Genres
We aim to break down each movie's genre information to better understand how individual genres contribute to a movie's success. Since movies can belong to multiple genres, expanding the dataset to reflect each genre individually will allow us to analyze the performance of each genre separately.

In [162]:
# Split genres into a list
final_table['genres'] = final_table['genres'].str.split(',')

# Explode the list into separate rows
exploded_table = final_table.explode('genres')

# Remove extra spaces around genre names
exploded_table['genres'] = exploded_table['genres'].str.strip()
exploded_table

genres                                        movie  \
0            Action & Adventure                                       avatar   
0                        Comedy                                       avatar   
0            Mystery & Suspense                                       avatar   
0     Science Fiction & Fantasy                                       avatar   
1            Action & Adventure  pirates of the caribbean: on stranger tides   
...                         ...                                          ...   
6223         Mystery & Suspense                                    following   
6226                     Comedy                            my date with drew   
6226                Documentary                            my date with drew   
6226           Special Interest                            my date with drew   
6226                 Television                            my date with drew   

      production_budget  worldwide_gross  ROI (%)  
0           425000000.0     2.776345e+09    553.0  
0           425000000.0     2.776345e+09    553.0  
0           425000000.0     2.776345e+09    553.0  
0           425000000.0     2.776345e+09    553.0  
1           410600000.0     1.045664e+09    155.0  
...                 ...              ...      ...  
6223             6000.0     2.404950e+05   3908.0  
6226             1100.0     1.810410e+05  16358.0  
6226             1100.0     1.810410e+05  16358.0  
6226             1100.0     1.810410e+05  16358.0  
6226             1100.0     1.810410e+05  16358.0  

[10909 rows x 5 columns]

Cleaning and Grouping by Genres
We aim to clean the data and group the movies by their genres to calculate the average Return on Investment (ROI) for each genre. Cleaning the data ensures that we remove any rows with missing or invalid information. Sorting the data by ROI will help us identify the top-performing genres, providing valuable information for future movie production decisions.

In [163]:
# Drop rows with NULLs in important columns
exploded_table = exploded_table.dropna(subset=['genres', 'movie', 'ROI (%)'])

# Group by 'genres' and calculate average ROI, production_budget, and worldwide_gross
grouped_table = exploded_table.groupby('genres').agg({
    'ROI (%)': 'mean',              # Average ROI
    'production_budget': 'mean',    # Average production_budget
    'worldwide_gross': 'mean'       # Average worldwide_gross
}).reset_index()

# Sort by ROI in descending order
grouped_table = grouped_table.sort_values(by='ROI (%)', ascending=False)

# Reset index to clean it up
grouped_table = grouped_table.reset_index(drop=True)

grouped_table.head()

genres      ROI (%)  production_budget  worldwide_gross
0       Cult Movies  4669.958333       2.545846e+07     8.253467e+07
1            Horror   975.298217       2.464406e+07     7.716784e+07
2  Special Interest   932.209677       1.217236e+07     3.201614e+07
3       Documentary   868.586466       1.323879e+07     3.199423e+07
4          Classics   839.873684       3.186409e+07     1.034252e+08

Normalise ROI
We normalize the ROI (%) values so they collectively sum to 100%. This helps us interpret each genre's contribution to the total ROI as a percentage share.

In [164]:
Total_Roi = grouped_table['ROI (%)'].sum()

# Normalize the ROI values to ensure they sum to 100%
grouped_table['Normalized_ROI'] = (grouped_table['ROI (%)'] / Total_Roi) * 100

# Ensure the normalized values sum exactly to 100 by adjusting based on total
normalized_total = grouped_table['Normalized_ROI'].sum()
adjustment_factor = 100 / normalized_total

# Re-adjust the normalized ROIs to ensure they sum exactly to 100
grouped_table['Normalized_ROI'] = grouped_table['Normalized_ROI'] * adjustment_factor
grouped_table

genres      ROI (%)  production_budget  \
0                 Cult Movies  4669.958333       2.545846e+07   
1                      Horror   975.298217       2.464406e+07   
2            Special Interest   932.209677       1.217236e+07   
3                 Documentary   868.586466       1.323879e+07   
4                    Classics   839.873684       3.186409e+07   
5            Sports & Fitness   740.619048       2.526810e+07   
6                  Television   695.370370       2.903874e+07   
7        Faith & Spirituality   652.086957       1.010326e+07   
8                   Animation   527.445783       8.096078e+07   
9   Musical & Performing Arts   509.462766       2.771714e+07   
10  Art House & International   481.818966       1.573763e+07   
11              Kids & Family   470.620985       6.421716e+07   
12         Mystery & Suspense   411.601010       3.389579e+07   
13  Science Fiction & Fantasy   410.494911       6.863892e+07   
14                    Romance   385.595156       2.982921e+07   
15                      Drama   317.076129       3.055283e+07   
16                     Comedy   309.905467       3.164979e+07   
17         Action & Adventure   281.021448       6.156793e+07   
18                    Western   197.364865       3.417673e+07   
19              Gay & Lesbian   171.000000       4.833333e+06   
20              Anime & Manga    53.000000       1.100000e+08   

    worldwide_gross  Normalized_ROI  
0      8.253467e+07       31.341139  
1      7.716784e+07        6.545445  
2      3.201614e+07        6.256269  
3      3.199423e+07        5.829279  
4      1.034252e+08        5.636581  
5      7.822937e+07        4.970461  
6      8.745679e+07        4.666787  
7      1.855955e+07        4.376302  
8      2.969315e+08        3.539807  
9      1.086628e+08        3.419119  
10     3.106344e+07        3.233595  
11     2.164677e+08        3.158443  
12     9.521296e+07        2.762347  
13     2.086122e+08        2.754924  
14     9.788020e+07        2.587816  
15     8.072119e+07        2.127969  
16     9.370029e+07        2.079845  
17     1.755505e+08        1.885998  
18     6.077859e+07        1.324560  
19     2.155033e+07        1.147619  
20     1.679107e+08        0.355695

Preparing Language and Popularity Data
By analyzing two key variables from The Movie DB dataset: original_language and popularity. The original_language column helps us understand which languages dominate box office performance, guiding strategic decisions about which language(s) to prioritize in production—either to reach the broadest audience or to tap into emerging markets. 
The popularity column offers insight into which films are most liked by audiences. By examining both these variables, we can uncover patterns between language, popularity, and potential box office future success.

In [165]:
tmdb_df.head(5)

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

We utilized the Data Wrangler tool to inspect the columns of interest: original_language and popularity. We confirmed that both columns had no missing values, ensuring that our analysis would proceed without any gaps in the data.

Preparing Directors and Foreign Gross Data
From our earlier data exploration, we have confirmed that there are no missing values in the relevant im.db tables and in the Numbers movie budgets dataset, which will allow us to proceed without needing additional imputation steps. Since these datasets originate from different sources and structures, we will need to carefully filter and merge them to ensure accuracy.

Extracting Director and Movie Title Data
First, we'll pull a clean list of directors and their movies from the IMDb database. Director names live in the 'persons' table, while movie titles are in 'movie_akas', and they're connected through ID references in the 'directors' table.

We're being extra careful to avoid duplicates by using DISTINCT in our queries, because we spotted multiple copies of the same director-movie pairs, probably due to how the data was originally entered. By filtering for unique entries and sticking to original titles, we're making sure our final list is clean and reliable.

In [166]:
# Connect to the database
conn = sqlite3.connect('Data/im.db')

# Updated query with DISTINCT
query = '''
SELECT DISTINCT
    p.primary_name AS director_name,
    ma.title AS movie_title,
    mb.start_year
FROM directors d
JOIN persons p ON d.person_id = p.person_id
JOIN movie_akas ma ON d.movie_id = ma.movie_id
JOIN movie_basics mb ON d.movie_id = mb.movie_id
WHERE ma.is_original_title = 1
'''

# Execute the query and load into a DataFrame
Directors_df = pd.read_sql_query(query, conn)

# Close connection
conn.close()
Directors_df.head()

director_name                                        movie_title  \
0   Colin Trevorrow                                     Jurassic World   
1    Andrew Stanton                                        John Carter   
2    Marc Jampolsky  Versailles Rediscovered - The Sun King's Vanis...   
3      Sam Zubrycki                      Miguelito - Canto a Borinquen   
4  Henning Beckhoff                                  Thing I Don't Get   

   start_year  
0        2015  
1        2012  
2        2019  
3        2019  
4        2018

Merging Directors with Worldwide Gross Data
Combining the director information with the worldwide gross earnings ensures that each movie in our dataset is matched with both its director and its revenue performance, setting the foundation for meaningful analysis.

In [167]:
# Rename for easy merge
Directors_df = Directors_df.rename(columns={'movie_title': 'movie'})

# bug fix: To resolve potential override and faulty merge
tn_df['movie'] = tn_df['movie'].str.lower()
Directors_df['movie'] = Directors_df['movie'].str.lower()

# Merge the two DataFrames
merged_df = pd.merge(tn_df[['movie', 'worldwide_gross']], Directors_df, on='movie', how='inner')

# Reorder columns
directors_fgross_df = merged_df[['movie', 'director_name', 'worldwide_gross', 'start_year']]

# Sort alphabetically by director name
directors_fgross_df = directors_fgross_df.sort_values(by='director_name').reset_index(drop=True)
directors_fgross_df

movie       director_name worldwide_gross  \
0                             akira     A.R. Murugadoss         $19,585   
1                             bully          Aaron Alon      $1,381,824   
2                          restless         Aaron Boltz      $2,772,511   
3                            circle          Aaron Hann         $10,024   
4     teen titans go! to the movies       Aaron Horvath     $51,620,593   
...                             ...                 ...             ...   
2443             along the roadside       Zoran Lisinac          $3,234   
2444                       prophecy       Zuri Rinpoche     $22,673,340   
2445         perfectos desconocidos  Álex de la Iglesia     $31,166,312   
2446                              9        Éric Tessier     $48,559,999   
2447                      ballerina          Éric Warin     $96,892,829   

      start_year  
0           2016  
1           2017  
2           2012  
3           2015  
4           2018  
...          ...  
2443        2013  
2444        2015  
2445        2017  
2446        2016  
2447        2016  

[2448 rows x 4 columns]

Aggregating Worldwide Gross Earnings by Director
We need to aggregate the data to eatablish for the fact that a single director may have multiple movies. This will give us a clearer picture of each director's overall financial impact.

In [168]:
# Remove commas, dollar sign and convert to float for calculation
directors_fgross_df['worldwide_gross'] = directors_fgross_df['worldwide_gross'].replace(r',', '', regex=True)
directors_fgross_df['worldwide_gross'] = directors_fgross_df['worldwide_gross'].replace(r'\$', '', regex=True).astype(float)

# Group by director_name and sum worldwide_gross
agg_directors_df = directors_fgross_df.groupby('director_name', as_index=False)['worldwide_gross'].sum()

# Sort by worldwide_gross in descending order
agg_directors_df = agg_directors_df.sort_values(by='worldwide_gross', ascending=False).reset_index(drop=True)

# Format numbers with commas 
agg_directors_df['worldwide_gross'] = agg_directors_df['worldwide_gross'].apply(lambda x: f"{int(x):,}")
agg_directors_df

director_name worldwide_gross
0           Anthony Russo   3,902,605,502
1               Joe Russo   3,902,605,502
2           Pierre Coffin   3,713,745,331
3       Christopher Nolan   3,086,180,484
4             Joss Whedon   2,992,084,614
...                   ...             ...
1683      Christian Sesma               0
1684      Deepak Rauniyar               0
1685  Russell Friedenberg               0
1686        Katie Aselton               0
1687        Brandon Trost               0

[1688 rows x 2 columns]

Preparing Content Rating and ROI Data
Understanding the relationship between a film’s content rating (such as G, PG, PG-13, R) and its Return on Investment (ROI) can provide valuable strategic guidance for the company’s new movie studio. Content ratings influence which audiences a movie can legally and culturally reach, and may affect production budgets, marketing strategies, and box office returns.

Merging Dataframes
To analyze the relationship between content ratings and ROI, we need to combine relevant information from two different sources. We will merge the final_table, which already contains ROI calculations and genre data, with the Rotten Tomatoes dataset (rt_df), which includes content rating information for each movie.

In [169]:
# computing the joined df
merged_df = final_table.merge(rt_df[['movie_title', 'content_rating']], left_on='movie', right_on='movie_title', how='left')
merged_df.sample(10)

genres  \
5817                                            [Drama]   
920   [Action & Adventure,  Comedy,  Kids & Family, ...   
1722               [Animation,  Comedy,  Kids & Family]   
432   [Comedy,  Documentary,  Drama,  Special Interest]   
2483                       [Drama,  Mystery & Suspense]   
3390               [Comedy,  Musical & Performing Arts]   
1938       [Action & Adventure,  Drama,  Kids & Family]   
4904  [Art House & International,  Comedy,  Drama,  ...   
424   [Action & Adventure,  Animation,  Comedy,  Kid...   
3624                                  [Drama,  Romance]   

                                   movie  production_budget  worldwide_gross  \
5817                          camp x-ray          1000000.0         101053.0   
920                      bedtime stories         80000000.0      221468935.0   
1722                           early man         50000000.0       45285890.0   
432                                 home         12000000.0              0.0   
2483               bringing out the dead         32000000.0       16640210.0   
3390  popstar: never stop never stopping         20000000.0        9537120.0   
1938                      the karate kid         40000000.0      351774938.0   
4904                    good bye, lenin!          6400000.0       79384539.0   
424                                 home        130000000.0      385997896.0   
3624                              hamlet          2000000.0        2419669.0   

          ROI (%)                         movie_title content_rating  
5817   -90.000000                          camp x-ray              R  
920    177.000000                     bedtime stories             PG  
1722    -9.000000                           early man             PG  
432   2985.333333                                home             NR  
2483   -48.000000               bringing out the dead              R  
3390   -52.000000  popstar: never stop never stopping              R  
1938   907.000000                      the karate kid             PG  
4904  1140.000000                    good bye, lenin!              R  
424   2985.333333                                home             NR  
3624   -19.500000                              hamlet          PG-13

Extracting Relevant Columns
We will extract only two columns: content_rating and ROI (%). These columns will allow us to analyze average returns across different rating categories.

In [170]:
# extracting the relevant columns
relevant_columns=["ROI (%)","content_rating"]
rel_joined_df = merged_df[relevant_columns].copy()
rel_joined_df.head(5)

ROI (%) content_rating
0    553.0          PG-13
1    155.0          PG-13
2    -57.0          PG-13
3    324.0          PG-13
4    583.0          PG-13

Handling Missing Values
We check for and drop rows with missing values in content_rating to ensure the dataset remains clean and usable.

In [171]:
# checking for missing_values
missing = rel_joined_df.isna().sum()
percent_missing = (missing / len(rel_joined_df))*100
percent_missing

ROI (%)           0.0
content_rating    0.0
dtype: float64

In [172]:
# dropping the rows with missing values
rel_joined_df.dropna(axis=0,subset=["content_rating"],inplace=True)
rel_joined_df.isna().sum()

ROI (%)           0
content_rating    0
dtype: int64

Aggregating ROI by Content Rating
Grouping the data by content_rating and computing the average ROI for each category. This will help reveal which content ratings are associated with higher financial returns.

In [173]:
# grouping the df
grouped_df = rel_joined_df.groupby("content_rating")["ROI (%)"].mean().reset_index().sort_values(by="ROI (%)",ascending=False)
grouped_df

content_rating     ROI (%)
0              G  751.944649
2             NR  572.694215
5              R  513.067253
3             PG  486.622322
4          PG-13  257.874652
1           NC17  125.000000

Further Exploratory Data Analysis
To ensure the data is as accurate and clean as possible we use further EDA to help us identify inconsistencies, outliers, and patterns that could distort results. Making the necessary adjustments at this stage improves the reliability of any statistical tests that follow.

Summarizing Dataframe
This initial step is important for understanding the completeness and quality of the data, identifying what types of analysis are appropriate, and planning any necessary cleaning or preprocessing before deeper exploration. In this section we will be summarizing the merged dataframe containing relevant columns.

In [174]:
# summary statistics of the cleaned dataframe
rel_joined_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6061 entries, 0 to 6060
Data columns (total 2 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ROI (%)         6061 non-null   float64
 1   content_rating  6061 non-null   object 
dtypes: float64(1), object(1)
memory usage: 142.1+ KB


The dataset has 6,061 rows and 2 columns, with no missing values. It contains both categorical and numerical columns, which determines the methods used for analysis

Describing Numerical Columns
This helps identify outliers, detect skewness, and understand how these variables compare in scale. It is essential for making informed analysis decisions, guiding further data exploration, and ensuring accurate modeling.

In [175]:
# describing the cleaned dataframe
rel_joined_df.describe()

ROI (%)
count   6061.000000
mean     448.614420
std     1817.558207
min     -100.000000
25%      -19.500000
50%      103.000000
75%      312.000000
max    49775.000000

Outliers Handling
Outliers often represent genuine blockbusters or flops that are essential to understanding the broader ROI trends across different content ratings. Removing these data points would have risked losing valuable insights. Additionally, keeping the outliers preserved the integrity of the original variable relationships. The code cell below was the simplest and most efficient way to handle the variables we had created, which are referenced in later analyses.

In [176]:
# grouping the df
cleaned_df2 = rel_joined_df.sort_values(by="ROI (%)",ascending=False)
cleaned_df2

ROI (%) content_rating
5993  49775.0              R
5930  43052.0              R
6020  41556.0              R
5883  41283.0              R
6035  34106.0          PG-13
...       ...            ...
4978   -100.0              R
5924   -100.0             NR
5926   -100.0              R
5927   -100.0             NR
5476   -100.0             NR

[6061 rows x 2 columns]